File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/grade/GradeDto.java
package com.side.tiggle.domain.grade;

import com.side.tiggle.domain.grade.model.Grade;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class GradeDto {
    long id;
    String name;
    String imageUrl;
    String description;

    public static GradeDto fromEntity(Grade grade) {
        GradeDto dto = new GradeDto();
        dto.id = grade.getId();
        dto.name = grade.getName();
        dto.imageUrl = grade.getImageUrl();
        dto.description = grade.getDescription();

        return dto;
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/grade/GradeDto.kt
package com.side.tiggle.domain.grade

import com.side.tiggle.domain.grade.model.Grade

data class GradeDto(
    val id: Long,
    val name: String,
    val imageUrl: String,
    val description: String
) {

    fun toEntity(): Grade {
        val grade = Grade (
            name = this.name,
            imageUrl = this.imageUrl,
            description = this.description
        )
        grade.id = this.id
        return grade
    }
    companion object {
        fun fromEntity(grade: Grade): GradeDto{
            return GradeDto(
                id = grade.id,
                name = grade.name,
                imageUrl = grade.imageUrl,
                description = grade.description
            )
        }
    }
}